src/server/game/Spells/SpellEffects.cpp

```diff
@@ -2232,6 +2232,15 @@ void Spell::EffectSummonType(SpellEffIndex effIndex)
             break;
     }
 
+    // Some creatures must be summoned as pets, yet have a different category or type set in SummonProperties.dbc.
+    // Until a generic way is found to handle them, replace the existing values.
+    switch (properties->Id)
+    {
+    case 628: // Hungry Plaguehound
+        properties->Category = SUMMON_CATEGORY_PET;
```

Comment:
Absolutely no idea, sorry. I see no reason it would cause any issue, though, as this happens before the creature itself is created.